### ISSUE TITLE
Missing property rootId for 'stripping' templateId

### ISSUE THREAD

**Issue by fervanrijswijk** [mdvanes/grunt-kot2html#3]
As we discussed. We need some sort of logic to 'strip' the template Id
`/*
- grunt-kot2html
- https://github.com/mdvanes/grunt-kot2html
  *
- Copyright (c) 2015 M.D. van Es
- Licensed under the MIT license.
  */

'use strict';

module.exports = function (grunt) {

```
grunt.registerMultiTask('kot2html', 'Grunt task to combine Knockout templates into one HTML for the Knockout.js-External-Template-Engine.', function () {

    var path = require('path'),
        files = grunt.file.expand(this.data.src),
        destObj = path.parse(this.data.dest),
        result = '',
        deepEqual = require('deep-equal'),
        prefix = this.data.prefix,
        idRoot = this.data.idRoot,
        relativePath;


    files.forEach(function (file) {
        grunt.verbose.writeln('Processing', file);


        // Prevent reading the output file
        if (!deepEqual(destObj, path.parse(file))) {

            relativePath = path.dirname(file);

            //check if a path is defined to used for stripping the base
            if(idRoot){
                if(path.dirname(file).indexOf(idRoot) === 0){
                    relativePath = relativePath.replace(idRoot, '');
                }else if(path.dirname(file).indexOf(idRoot) > 0){
                    throw new Error('idRoot is not the root of the source-file');
                }else if(path.dirname(file).indexOf(idRoot) === -1){
                    throw new Error('idRoot does exist in root path of source-file');
                }
            }

            // Strip the extension to determine the template name
            var templateId = prefix +
                relativePath + '/' +
                path.basename(file).replace('.html', '');
            var content = grunt.file.read(file);

            result += '<script type="text/html" id="' + templateId + '">\n    ' + content + '\n</script>\n\n';
        }
    });

    grunt.file.write(this.data.dest, result);
    grunt.log.writeln('Compilation to ' + this.data.dest + ' completed successfully');
});
```

};
`

**Comment by mdvanes** [comment]
The suggestion is that there should be an arbitrary root for the template ids.

E.g. at this moment, the template location is provided in the src property. Imagine this structure:
- _/_ - git root
  - _themes_
    - _mytheme_ - project root with Gruntfile and package.json
      - _template_ - templates dir

The kot2html config in the Gruntfile will contain:

```
{
    prefix: 'themes/mytheme/',
    src: 'template/**/*.html',
    dest: 'generated-templates.html'
}
```

This will generate a generated-templates.html in the dir "theme/mytheme" that will contain template snippets like:

```
<script type="text/html" id="themes/mytheme/template/li">
    <li></li>
</script>
```

The id is a composite of prefix + src + path.

The suggestion by @fervanrijswijk is to solve the situation where the structure is:
- _/_ - git root, also the project root with Gruntfile and package.json
  - _dev_ - a dir that should not be in the template id, because it is not distributed
    - _themes_
      - _mytheme_
        - _template_ - templates dir

with this config:

```
{
    prefix: '',
    src: 'dev/themes/mytheme/template/**/*.html',
    dest: 'generated-templates.html'
}
```

This will generate a snippet like:

```
<script type="text/html" id="dev/themes/mytheme/template/li">
    <li></li>
</script>
```

so the ID will contain the "dev" dir, but this dir is not to be in the final distribution package.

**Comment by mdvanes** [comment]
a suggested solution is to add a property that reflects the part of the path that will be skipped in setting the template id

```
{
    prefix: '',
    srcRoot: 'dev/',
    src: 'themes/mytheme/template/**/*.html',
    dest: 'generated-templates.html'
}
```

which should yield an id like: `themes/mytheme/template/li`

**Comment by mdvanes** [comment]
Should be fixed in v0.3.1
